File: app/src/main/java/ru/application/homemedkit/helpers/ImageHelper.java
package ru.application.homemedkit.helpers;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;

import androidx.appcompat.content.res.AppCompatResources;

import java.util.Arrays;

import ru.application.homemedkit.R;

public class ImageHelper {
    public static Drawable getIconType(Context context, String form) {
        TypedArray icons = context.getResources().obtainTypedArray(R.array.medicine_types_icons);
        String[] types = context.getResources().getStringArray(R.array.medicine_types);
        String type = StringHelper.formName(form).toUpperCase();
        Drawable icon = AppCompatResources.getDrawable(context, R.drawable.vector_type_unknown);

        int index = Arrays.asList(types).indexOf(type);
        if (index != -1) icon = icons.getDrawable(index);

        return icon;
    }
}

File: app/src/main/java/ru/application/homemedkit/helpers/ImageHelper.kt
package ru.application.homemedkit.helpers

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.drawable.Drawable
import androidx.appcompat.content.res.AppCompatResources
import ru.application.homemedkit.R

@SuppressLint("Recycle")
fun getIconType(context: Context, form: String): Drawable {
    val icons = context.resources.obtainTypedArray(R.array.medicine_types_icons)
    val types = context.resources.getStringArray(R.array.medicine_types)
    val type = formName(form).uppercase()

    return when (val index = types.indexOf(type)) {
        -1 -> AppCompatResources.getDrawable(context, R.drawable.vector_type_unknown)!!
        else -> icons.getDrawable(index)!!
    }
}